packages/core/test/stylable-processor.spec.ts

```diff
@@ -305,6 +306,31 @@ describe('Stylable postcss process', () => {
         expect(Object.keys(result.elements).length).to.eql(1);
     });
 
+    it('should not collect typed elements or classes in unknown functional selectors', () => {
+        const result = processSource(
+            `
+            :unknown(Unknown.unknown) {}
+            ${reservedPseudoClasses.map((name) =>
+                name.startsWith(`nth`)
+                    ? `:${name}(5n, El-${name}.cls-${name}) {}`
+                    : `:${name}(El-${name}.cls-${name}) {}`
+            )}
+        `,
+            { from: 'path/to/style.css' }
+        );
+
+        expect(Object.keys(result.elements)).to.not.include(`Unknown`);
+        expect(Object.keys(result.classes)).to.not.include(`unknown`);
+        expect(Object.keys(result.elements)).to.not.include(`El-global`);
+        expect(Object.keys(result.classes)).to.not.include(`cls-global`);
+
+        const allWithoutGlobal = reservedPseudoClasses.filter((name) => name !== `global`);
```

Comment:
not sure I understand the purpose of this set of tests

Reply:
It checks that any nested selectors within the known functional pseudo-class list are collected. With the exception of `:global` which Stylable doesn't pick selectors from.